Sources/X509/CertificatePrivateKey.swift

```diff
@@ -378,3 +378,43 @@ extension Certificate.PrivateKey {
         }
     }
 }
+
+@available(macOS 10.15, iOS 13, watchOS 6, tvOS 13, macCatalyst 13, visionOS 1.0, *)
+extension Certificate.PrivateKey {
+    /// Return a list of all supported signature types for this private key. The ordering is not a comment on the
+    /// preference or security of the contained algorithms.
+    @inlinable
+    public func supportedSignatureAlgorithms(includeLegacyAlgorithms: Bool = false) -> [Certificate.SignatureAlgorithm]
```

Comment:
Instead of "legacy", the term used internally in this repo, and in public API in Swift Crypto, is "insecure".

E.g. https://github.com/apple/swift-certificates/blob/9ad2ac038e110470708cc670c27460ce0d93cfaa/Sources/X509/Digests.swift#L25

However, since Swift Certificates' public API has a value for `Certificate.SignatureAlgorithm.sha1WithRSAEncryption`, that makes no such distinction, I don't know if we want have this flag or not in the API.

WDYT?